clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java

```diff
@@ -443,145 +441,120 @@ private boolean alreadyHaveToken(String newToken, PushType pushType) {
         return alreadyAvailable;
     }
 
-    public void runInstanceJobWork(final Context context, final JobParameters parameters) {
-        Task<Void> task = CTExecutorFactory.executors(config).postAsyncSafelyTask();
-        task.execute("runningJobService", new Callable<Void>() {
-            @Override
-            public Void call() {
-                if (!isNotificationSupported()) {
-                    Logger.v(config.getAccountId(), "Token is not present, not running the Job");
-                    return null;
-                }
-
-                Calendar now = Calendar.getInstance();
-
-                int hour = now.get(Calendar.HOUR_OF_DAY); // Get hour in 24 hour format
-                int minute = now.get(Calendar.MINUTE);
-
-                Date currentTime = parseTimeToDate(hour + ":" + minute);
-                Date startTime = parseTimeToDate(Constants.DND_START);
-                Date endTime = parseTimeToDate(Constants.DND_STOP);
+    public void runPushAmpWork(final Context context) {
+        Logger.v(config.getAccountId(), "Pushamp - Running work request");
+        if (!isNotificationSupported()) {
+            Logger.v(config.getAccountId(), "Pushamp - Token is not present, not running the work request");
+            return;
+        }
 
-                if (isTimeBetweenDNDTime(startTime, endTime, currentTime)) {
-                    Logger.v(config.getAccountId(), "Job Service won't run in default DND hours");
-                    return null;
-                }
+        Calendar now = Calendar.getInstance();
 
-                long lastTS = baseDatabaseManager.loadDBAdapter(context).getLastUninstallTimestamp();
+        int hour = now.get(Calendar.HOUR_OF_DAY); // Get hour in 24 hour format
+        int minute = now.get(Calendar.MINUTE);
 
-                if (lastTS == 0 || lastTS > System.currentTimeMillis() - 24 * 60 * 60 * 1000) {
-                    try {
-                        JSONObject eventObject = new JSONObject();
-                        eventObject.put("bk", 1);
-                        analyticsManager.sendPingEvent(eventObject);
+        Date currentTime = parseTimeToDate(hour + ":" + minute);
+        Date startTime = parseTimeToDate(Constants.DND_START);
+        Date endTime = parseTimeToDate(Constants.DND_STOP);
```

Comment:
We should not create new date object everytime.

Reply:
As in the startTime and the endTime can be Date constants itself rather than having DND_START and DND_STOP?

Reply:
we cannot have them as fixed constants i guess because we have to compute them for daily timestamps of DND hours. But the SDF object can be reused, you can explore if there are better ways to do this

Reply:
This has been addressed